### PR TITLE
Refactor type names to improve Kapacitor documentation.

### PR DESCRIPTION
**Pull request by jonseymour**
As discussed in #433 the type names used in the online Kapacitor documentation are unhelpful since the correspondence between Tick script keywords and type names is not as clear as it could be.

Since this documentation is auto-generated we really need to fix the type names in the source code.

Since this would be tedious and error prone if done manually, the refactoring is done automatically
with a bash script which:

- applies one refactoring at a time with grep and sed
- compiles the resulting code to check for compile errors
- commits the result to git
- finally, removes itself from the code base.

Once this is ready to be merged, the first commit in the series should be rebased onto the current origin/master and refactor-doc.sh executed which will produce a refactoring that exactly matches the current origin/master.

The other commits in this series are provided so that the refactorings the script is doing can be reviewed.

/cc @nathanielc

### CODE REVIEW

File: pipeline/batch.go [hunk 1]
```diff
@@ -129,8 +129,8 @@ type BatchNode struct {
 	Cluster string
 }
 
-func newBatchNode() *BatchNode {
-	b := &BatchNode{
+func newQueryNode() *QueryNode {
+	b := &QueryNode{
 		chainnode: newBasicChainNode("batch", BatchEdge, BatchEdge),
```
Comment (nathanielc) [inline pipeline/batch.go]: Change "batch" to "query"

File: pipeline/stream.go [hunk 2]
```diff
@@ -104,8 +104,8 @@ type StreamNode struct {
 	Truncate time.Duration
 }
 
-func newStreamNode() *StreamNode {
-	return &StreamNode{
+func newFromNode() *FromNode {
+	return &FromNode{
 		chainnode: newBasicChainNode("stream", StreamEdge, StreamEdge),
```
Comment (nathanielc) [inline pipeline/stream.go]: Change "stream" to "from"

**Comment by nathanielc** [comment]
@jonseymour Thanks! The original names have drifted in meaning, I really like these new names simple and clear.

There are a few spots that should be changed to reflect this name change.

* https://github.com/influxdata/kapacitor/blob/master/task_master.go#L509 change srcstream0 to stream0. 
* Two other places in the diff that I commented.

File: pipeline/batch.go [hunk 3]
```diff
@@ -21,12 +21,12 @@ import (
...
+type BatchNode struct {
 	node
 }
 
-func newSourceBatchNode() *SourceBatchNode {
-	return &SourceBatchNode{
+func newBatchNode() *BatchNode {
+	return &BatchNode{
 		node: node{
 			desc:     "srcbatch",
```
Comment (nathanielc) [inline pipeline/batch.go]: This should also change to "batch"

File: pipeline/stream.go [hunk 4]
```diff
@@ -6,17 +6,17 @@ import (
...
+type StreamNode struct {
 	node
 }
 
-func newSourceStreamNode() *SourceStreamNode {
-	return &SourceStreamNode{
+func newStreamNode() *StreamNode {
+	return &StreamNode{
 		node: node{
 			desc:     "srcstream",
```
Comment (nathanielc) [inline pipeline/stream.go]: This should change to "stream"

**Comment by nathanielc** [comment]
@jonseymour I would like to get this change in before too much changes and makes this PR invalid. Do you think you will have time to make a few small changes and rebase over the next few days?

**Comment by nathanielc** [comment]
closing in favor of #476

**Comment by jonseymour** [comment]
@nathanielc apologies for not getting back to you earlier about this change - I have deeply absorbed in some other tasks that reduced my time for working on these changes. Thanks for making the changes in #476

**Comment by nathanielc** [comment]
@jonseymour No problem, Kapacitor is cutting a release and I wanted these good changes to make it in, so I finished up. Thanks for getting it started!

**Pull request closed without merging**